sa/migrations.sh

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if type realpath >/dev/null 2>&1 ; then
+  cd "$(realpath -- $(dirname -- "$0"))"
+fi
+
+# posix compliant escape sequence
+esc=$'\033'"["
+
+#
+# Defaults
+#
+DB_NEXT_PATH="_db-next/migrations"
+OUTCOME="ERROR"
+PROMOTE=()
+RUN=()
+
+#
+# Print Functions
+#
+function print_outcome() {
+  if [ "$OUTCOME" == OK ]
+  then
+    echo -e "${esc}0;32;1m""$OUTCOME""${esc}0m"
+  else
+    echo -e "${esc}0;31;1m""$OUTCOME""${esc}0m"
+  fi
+}
+
+function print_usage_exit() {
+  echo "$USAGE"
+  exit 0
+}
+
+function print_heading() {
+  echo
+  echo -e "${esc}0;34;1m"$1"${esc}0m"
+}
+
+function print_from() {
+  echo -e "from: ${esc}0;36;1m"$1"${esc}0m"
+}
+
+function print_to() {
+  echo -e "to:   ${esc}0;32;1m"$1"${esc}0m"
+}
+
+#
+# CLI Helper Functions
+#
+function check_arg() {
+  if [ -z "$OPTARG" ]
+  then
+    exit_msg "No arg for --$OPT option, use: -h for help">&2
+  fi
+}
+
+function exit_msg() {
+  # complain to STDERR and exit with error
+  echo "$*" >&2
+  exit 2
+}
+
+#
+# Utility Functions
+#
+function get_migrations() {
+  migrations=( $(find $DB_NEXT_PATH -mindepth 1 -maxdepth 1 -not -path '*/\.*' -type "$1"  2> /dev/null | sort) )
+  if [ -z "${migrations[@]+x}" ]
+  then
+    echo "There are no migrations at path: "\"$DB_NEXT_PATH\"""
+    exit 1
+  fi
+}
+
+function print_migrations(){
+  iter=1
+  for mig_file in "${migrations[@]}"
+  do
+    echo ""$iter") $(basename -- "${mig_file}")"
+    iter=$(expr "$iter" + 1)
+  done
+}
+
+#
+# Main CLI Parser
+#
+USAGE="$(cat -- <<-EOM
+
+Usage:
+Boulder DB Migrations CLI:
+
+  Helper for listing, promoting, and demoting Boulder schema files
+
+  ./$(basename "${0}") [OPTION]...
+
+  -l, --list-next               Lists schemas exclusively present in sa/db-next
+  -c, --list-current            Lists schemas exclusively present in sa/db
```

Comment:
It looks to me like the implementation of this is "scan the _db-next folder for symlinks, print those". So this is actually listing schemas which exist in both db and db-next, not those that are present _exclusively_ in db (which, there shouldn't be any, but if there were, this wouldn't find them).

Reply:
yeah, I can change the wording here